plugins/dev9ghzdrk/pcap_io.cpp

```diff
@@ -51,29 +51,43 @@ int GetMACAddress(char *adapter, mac_address* addr)
 {
 	int retval = 0;
 #ifdef _WIN32
-	static IP_ADAPTER_INFO AdapterInfo[128];       // Allocate information
-												 // for up to 128 NICs
-	static PIP_ADAPTER_INFO pAdapterInfo;
-	ULONG dwBufLen = sizeof(AdapterInfo);	// Save memory size of buffer
-
-	DWORD dwStatus = GetAdaptersInfo(      // Call GetAdapterInfo
-	AdapterInfo,                 // [out] buffer to receive data
-	&dwBufLen);                  // [in] size of receive data buffer
-	if(dwStatus != ERROR_SUCCESS)    // Verify return value is
-		return 0;                       // valid, no buffer overflow
-
-	pAdapterInfo = AdapterInfo; // Contains pointer to
-											   // current adapter info
+	static IP_ADAPTER_ADDRESSES AdapterInfo[128];
+
+	static PIP_ADAPTER_ADDRESSES pAdapterInfo;
+	ULONG dwBufLen = sizeof(AdapterInfo);
+
+	DWORD dwStatus = GetAdaptersAddresses(
+		AF_UNSPEC,
+		GAA_FLAG_INCLUDE_PREFIX,
+		NULL,
+		AdapterInfo,
+		&dwBufLen
+	);
+	if(dwStatus != ERROR_SUCCESS)
+		return 0;
+
+	pAdapterInfo = AdapterInfo;
+
+	char adapter_desc[128] = "\0";
+
+	// Must get friendly description from the cryptic adapter name
+	for (int ii = 0; ii < pcap_io_get_dev_num(); ii++)
+		if (0 == strcmp(pcap_io_get_dev_name(ii), adapter) )
+			strcpy(adapter_desc, pcap_io_get_dev_desc(ii));
```

Comment:
Shouldn't there be a `break;` after a successful match?

Reply:
Optional here. There can only be one match because Windows doesn't allow duplicate descriptions

Reply:
It's more about making the intention of the code clearer. If someone reads this code they might be confused about why the search hasn't ended after a match has occurred.